### ISSUE TITLE
Custom dashboard suddenly not working anymore

### ISSUE THREAD

**Issue by yoorshop** [firehol/netdata#1873]
Title displayed, and blank page for rest

tried several backup from this html file

**Comment by yoorshop** [comment]
Custom dashboard is on external server
Netdata local works perfectly, udpated last, disabled firewalls, on both servers, flushed blocked  IPs too

**Comment by yoorshop** [comment]
now is back, weird

**Comment by ktsaou** [comment]
When this happens, press F12 on your browser and check what fails.

**Comment by PacMask** [comment]
I have the same problem. I have a pfsense box where i host netdata, but i want to host a html file with all the charts on my apache-server which is run under Windows on the same network. I get full local experience, but when try to access it from the Internet, it just gives a blank page. I get no related errors when i inspect the page either. I think it's something with loading dashboard.js but i'm too little unexperienced to troubleshoot further. Any tips for investigating further?

**Comment by ktsaou** [comment]
On the browser press F12 and switch to the network tab. If there are no errors, reload the page (hit F5). You will see all the netdata files loading, like this:

![image](https://cloud.githubusercontent.com/assets/2662304/23486614/d7e51f48-fee9-11e6-94a3-addab0c78db2.png)

The status column is of interest. `200` means OK. You should have non-200 statuses there.

**Comment by l2isbad** [comment]
@PacMask 
what about
```java 
<script type="text/javascript">var netdataServer = "http://your.netdata.server:19999";</script>
```
and
```java
<div data-netdata="unique.id"
     data-host="http://another.netdata.server:19999/"
     ></div>
```

Can you access `netdata.server:19999` from internet?

**Comment by PacMask** [comment]
@ktsaou I get errors loading the dashboard.js as i suspected: http://i.imgur.com/QoVkrQ5.png

@l2isbad I dont really understand, should i use "script type" in the <head> of the page im hosting, or under the <script>?

I cannot access my netdata server from the DNS name, since its on my router. But should i use the local IP address for all servers, or the DNS name in the config files?

This is my current hosted html-file (changed the port to 1994):
`

<head>
    <title>Dashboard</title>
    <meta name="application-name" content="netdata">

    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

    <meta property="og:locale" content="en_US" />
    <meta property="og:image" content="https://cloud.githubusercontent.com/assets/2662304/22945737/e98cd0c6-f2fd-11e6-96f1-5501934b0955.png"/>
    <meta property="og:url" content="http://my-netdata.io/"/>
    <meta property="og:type" content="website"/>
    <meta property="og:site_name" content="netdata"/>
    <meta property="og:title" content="netdata - real-time performance monitoring, done right!"/>
    <meta property="og:description" content="Stunning real-time dashboards, blazingly fast and extremely interactive. Zero configuration, zero dependencies, zero maintenance." />
    
</head>
<script>
// this section has to appear before loading dashboard.js

// Select a theme.
// uncomment on of the two themes:

// var netdataTheme = 'default'; // this is white
var netdataTheme = 'slate'; // this is dark


// Set the default netdata server.
// on charts without a 'data-host', this one will be used.
// the default is the server that dashboard.js is downloaded from.

var netdataServer = 'http://192.168.0.1:1994/';
</script>



<script type="text/javascript" src="http://192.168.0.1:1994/dashboard.js"></script> 


`

**Comment by ktsaou** [comment]
This is the problem:

```
var netdataServer = 'http://192.168.0.1:1994/';
```

192.168.0.1 cannot be accessed from the internet. Your web browser attempts to connect to this host.

**Comment by graphex** [comment]
I've suddenly had a dashboard stop working as well. There is just nothing displayed where graphs should be, and the divs of the graphs are collapsed into 1px high.

All files are loading without error, and the same HTML and CSS works in a graph page for a different, slightly older cluster which has older versions of netdata. In other words, my custom dashboard hasn't changed, but it is hitting machines with slightly newer versions of netdata now, and the graphs won't render.

Going directly to the netdata page (i.e. http://x.x.x.x:19999/) works just fine.

The javascript console doesn't give any indication of errors in the script:
```
>NETDATA.errorLast
Object {code: 0, message: "You are doing fine!", datetime: 0}
```
I am using bootstrap.css with surrounding areas of the page, and I tried updating to the most recent version of bootstrap (3.3.7 as of now) and it did not make the graphs visible again.

**Comment by ktsaou** [comment]
Hi. Can you give me something (i.e. a sample custom dashboard) that reproduces this?

**Comment by graphex** [comment]
I found the fix for my problem. I needed to call `NETDATA.start()` on my dashboard page after my graphs. I'm not sure if the behavior intentionally changed in the newer versions I started using, or it wasn't supposed to have worked before, but it didn't previously require me to make that explicit call to at least see graphs render. On the older versions, I often had to manually reload the page to get some of the graphs to render, and now they seem to reliably render, so it does seem like an improvement to explicitly call `start()`, I just didn't have much of a cue that it was necessary. Probably PEBCAK.
